appinventor/components/src/com/google/appinventor/components/runtime/Canvas.java

```diff
@@ -445,7 +447,7 @@ private Bitmap buildCache() {
         int height = getHeight();
         cache = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
         android.graphics.Canvas c = new android.graphics.Canvas(cache);
-        layout(0, 0, width, height);
+        //layout(0, 0, width, height);
```

Comment:
What exactly is the reasoning behind removing the layout call?

Reply:
According to my findings, I understood that ```layout``` function is used to for positioning and sizing of the views in the viewgroups. So I suspect that this function call may resetting the positioning of the image. So i tried removing this call and test the apk, than it worked perfectly.

Reply:
I'm not sure why it would be resetting the position of the image though. Every ImageSprite has an X Y coordinate and that should not change due to this layout call. Either it was in the right spot and then ends up in the wrong spot after this call, or it's in the wrong spot and you're now measuring that. In the former case, we should figure out why the sprites are ending up in the wrong positions as that could be indicative of a bug elsewhere in the code.

Reply:
Can you tell me the use of the layout call when building the cache?